internal/strategy/platform_service_strategy.go

```diff
@@ -0,0 +1,160 @@
+	//return "", nil, fiber.NewError(fiber.StatusBadRequest, "Platform Service does not support download in Open Source mode")
+}
+
+func (s *PlatformServiceStrategy) DownloadChefPlatform(params *omnitruck.RequestParams) (url string, respBody io.ReadCloser, header http.Header, msg string, code int, err error) {
+	resp := clients.Response{}
+	request := s.LicenseClient.GetReplicatedCustomerEmail(params.LicenseId, s.LicenseServiceUrl, &resp)
+
+	if !request.Ok {
+		s.Log.Errorf("Received error response from getReplicatedCustomer")
+		return "", nil, nil, request.Message, request.Code, fmt.Errorf("%s Error while fetching replicated customer email: %s", request.Code, request.Message)
+		//return "", nil, fmt.Errorf("%s Error while fetching replicated customer email: %s", request.Code, request.Message)
+	}
+
+	var replicatedEmailResp clients.GetReplicatedCustomerResponse
+	err = JsonUnmarshal(request.Body, &replicatedEmailResp)
+
+	if err != nil {
+		s.Log.Errorf("Error while unmarshalling getReplicatedCustomer response : %s", err.Error())
+		return "", nil, nil, constants.UNMARSHAL_ERR_MSG, http.StatusInternalServerError, fmt.Errorf("%s Error while fetching replicated customer email: %s", request.Code, request.Message)
+		//return server.SendErrorResponse(c, http.StatusInternalServerError, constants.UNMARSHAL_ERR_MSG+err.Error())
+	}
+
+	if replicatedEmailResp.StatusCode != http.StatusOK {
+		s.Log.Errorf("Received error response from getReplicatedCustomer")
+		return "", nil, nil, replicatedEmailResp.Message, replicatedEmailResp.StatusCode, fmt.Errorf("%s Error while fetching replicated customer email: %s", replicatedEmailResp.StatusCode, replicatedEmailResp.Message)
+		//return server.SendErrorResponse(c, replicatedEmailResp.StatusCode, replicatedEmailResp.Message)
+
+	}
+	s.Log.Debug("Successfully fetched replicated customer email")
+
+	//2. Run a search customer on replicated with email
+	requestId := s.locals["requestId"].(string)
+	customers, err := s.Replicated.SearchCustomersByEmail(replicatedEmailResp.ReplicatedEmail, requestId)
+
+	if err != nil {
+		s.Log.Errorf("Error while fetching replicated customers with Email : %s", err.Error())
+		return "", nil, nil, constants.REPLICATED_CUSTOMER_ERROR, http.StatusInternalServerError, fmt.Errorf("%s Error while fetching replicated customers with Email: %s", http.StatusInternalServerError, constants.REPLICATED_CUSTOMER_ERROR)
+		//return server.SendErrorResponse(c, http.StatusInternalServerError, constants.REPLICATED_CUSTOMER_ERROR)
+	}
+
+	if len(customers) == 0 {
+		s.Log.Errorf("No replicated customers found with Email : %s", replicatedEmailResp.ReplicatedEmail)
+		return "", nil, nil, constants.REPLICATED_CUSTOMER_ERROR, http.StatusInternalServerError, fmt.Errorf("%s No replicated customers found with Email: %s", http.StatusInternalServerError, constants.REPLICATED_CUSTOMER_ERROR)
```

Comment:
Please resolve the sonarqube errors: `fmt.Errorf format %s has arg http.StatusInternalServerError of wrong type int`